main.py

```diff
@@ -1,65 +1,340 @@
+    out_row[5] = address_components['components']['output_address']
+    out_row[12] = address_components['components']['mailing']['zipcode']
+    if out_row[12]:
+        geocode_stats['zip'] += 1
+
+    geocode_stats['total'] += 1
+
+    ## No GPS lat/lon
+    if row[9] == None:
+        if address_components['components']['cl_seg_id'] != None and \
+           address_components['components']['cl_seg_id'] in centriods:
+            lat_lon = centriods[address_components['components']['cl_seg_id']]
+            out_row[9] = lat_lon['Lat']
+            out_row[10] = lat_lon['lon']
+            out_row[11] = False
+            geocode_stats['success'] += 1
+        else:
+            if address_components['components']['cl_seg_id'] != None:
+                geocode_stats['failed_segment'] += 1
+                segment = address_components['components']['cl_seg_id']
+            else:
+                geocode_stats['failed_address'] += 1
+                segment = None
+
+            if segment:
+                geocode_stats['failed_segments'].add(segment)
+            logger.info('Geocode - {} not found'.format(segment or row[5]))
     else:
-        return division
+        geocode_stats['gps'] += 1
+        out_row[11] = True
 
-def fix_prisons_agency (issuing_agency):
-    if issuing_agency == 'RED LI':
-        return 'PRISONS'
+    return out_row
+
+def load_index_file(path, _type):
+    if path == None:
+        return {}
+
+    with smart_open(path, 'r') as file:
+        reader = csv.DictReader(file)
+
+        index = {}
+        for row in reader:
+            if _type == 'plates_file':
+                index[row['state'] + row['plate']] = row
+            elif _type == 'license_file':
+                index[row['ticket_number']] = row['anon_ticket_number']
+            elif _type == 'centriod_file':
+                index[row['SEG_ID']] = row
+            else:
+                raise Exception('`{}` not a supported index file type'.format(_type))
+
+        return index
+
+def save_index_file(path, index, headers, _type):
+    if path:
+        with smart_open(path, 'w') as file:
+            writer = csv.DictWriter(file, fieldnames=headers)
+
+            writer.writeheader()
+
+            for key, value in index.items():
+                if _type == 'plates_file':
+                    writer.writerow(value)
+                elif _type == 'license_file':
+                    writer.writerow({'ticket_number': key, 'anon_ticket_number': value})
+                else:
+                    raise Exception('`{}` not a supported index file type'.format(_type))
+
+@click.command()
+@click.option('--plates-file')
+@click.option('--ticket-numbers-file')
+@click.option('--centriod-file')
+@click.option('--latlon-input/--no-latlon-input', is_flag=True, default=True)
+@click.option('--deduplicate/--no-deduplicate', is_flag=True, default=False, help="Add 'E' to the end of duplicate ticket numbers")
```

Comment:
What's the purpose of this? I assume this is a new thing that came up last week in your convo w/Jerry. Maybe a comment giving context?